### DISCUSSION TITLE
Rust project in sub dir

### DISCUSSION THREAD

**Discussion by eitsupi** [oxsecurity/megalinter#2461] [Q&A]
We need a command like `cargo clippy --manifest-path sub/dir/Cargo.toml` to inspect Rust projects in subdirectories with clippy, is there any way to make something like this work with MegaLinter?

I tried the following in `.mega-linter.yml`, but it does not seem working.

```yml
RUST_CLIPPY_ARGUMENTS:
  - --manifest-path sub/dir/Cargo.toml
```

**Comment by nvuillam** [comment]
Arguments must be separated :)
Please can you try the following config:

```yaml
RUST_CLIPPY_ARGUMENTS:
  - --manifest-path
  - sub/dir/Cargo.toml
```

If u add `LOG_LEVEL: DEBUG` and inspect the log, you'll see the exact command line called by MegaLinter :)

**Reply by eitsupi** [reply]
Thanks, but it doesn't seem working too.

```log
   --Error detail:
  error: unexpected argument 'sub/dir/Cargo.toml' found
```

**Reply by nvuillam** [reply]
When you activate fine trace with LOG_LEVEL: DEBUG, what is the command line used ?

**Reply by eitsupi** [reply]
```log
✅ Linted [REPOSITORY] files with [git_diff] successfully - (0.01s) (expand for details)
[Text Reporter] Generated TEXT report: /github/workspace/megalinter-reports/linters_logs/SUCCESS-REPOSITORY_GIT_DIFF.log
[clippy] command: ['cargo-clippy', '--manifest-path', 'sub/dir/Cargo.toml']
[clippy] CWD: /github/workspace
[clippy] result: 1 error: unexpected argument 'sub/dir/Cargo.toml' found
```

Maybe `project/sub/dir/Cargo.toml` shoud be used?

**Reply by Kurt-von-Laven** [reply]
Note that the current working directory it is running in is `/github/workspace`, so that should settle any questions of which relative path is the correct one.

**Reply by eitsupi** [reply]
But that seems quite strange considering that `.mega-linter.yml` could be referenced outside of GitHub.

**Reply by Kurt-von-Laven** [reply]
Are you running it in GitHub Actions though?
